apps/game_backend/lib/game_backend/curse_of_mirra/quests.ex

```diff
@@ -157,52 +162,45 @@ defmodule GameBackend.CurseOfMirra.Quests do
     |> Repo.transaction()
   end
 
-  def add_daily_quests_to_user_id(user_id, amount) do
+  def add_quests_to_user_id_by_type(user_id, type, amount) do
     available_quests =
-      get_user_missing_quests_by_type(user_id, "daily")
+      get_user_missing_quests_by_type(user_id, type)
       |> Enum.shuffle()
 
-    if amount > Enum.count(available_quests) do
-      {:error, :not_enough_quests_in_config}
-    else
-      {multi, _quests} =
-        Enum.reduce(1..amount, {Multi.new(), available_quests}, fn
-          _index, {multi, [quest | next_quests]} ->
-            attrs = %{
-              user_id: user_id,
-              quest_id: quest.id,
-              status: "available"
-            }
+    amount_of_quest_available = Enum.count(available_quests)
 
-            changeset = UserQuest.changeset(%UserQuest{}, attrs)
+    cond do
+      type not in Ecto.Enum.values(GameBackend.Quests.Quest, :type) ->
+        {:error, :quest_type_not_implemented}
 
-            multi = Multi.insert(multi, {:insert_user_quest, user_id, quest.id}, changeset)
+      amount > amount_of_quest_available ->
+        {:error, "Not enough quests, requested: #{amount} available: #{amount_of_quest_available}"}
 
-            {multi, next_quests}
-        end)
+      true ->
+        {multi, _quests} =
+          Enum.reduce(1..amount, {Multi.new(), available_quests}, fn
+            _index, {multi, [quest | next_quests]} ->
+              attrs = %{
+                user_id: user_id,
+                quest_id: quest.id,
+                status: "available"
+              }
 
-      Repo.transaction(multi)
-    end
-  end
+              changeset = UserQuest.changeset(%UserQuest{}, attrs)
 
-  def get_user_daily_quests_completed(%User{
-        arena_match_results: arena_match_results,
-        user_quests: user_quests
-      }) do
-    user_quests
-    |> Enum.reduce([], fn user_quest, acc ->
-      if completed_quest?(user_quest, arena_match_results) and user_quest.quest.type == "daily" do
-        [user_quest | acc]
-      else
-        acc
-      end
-    end)
+              multi = Multi.insert(multi, {:insert_user_quest, user_id, quest.id}, changeset)
+
+              {multi, next_quests}
+          end)
+
+        Repo.transaction(multi)
+    end
   end
 
   def reroll_daily_quest(daily_quest_id) do
     reroll_configurations = Application.get_env(:game_backend, :quest_reroll_config)
 
-    daily_quest =
+    {:ok, daily_quest} =
```

Comment:
This will break if :error is returned, are we ok with that?

Reply:
hmmmm i'd rather not let this break here, i'll move the daily quest retrieval to the controller just as anything else